test/dynamo/test_hooks.py

```diff
@@ -316,17 +320,55 @@ def forward(self, x):
         aot_out[0].backward(torch.ones(4))
 
         x2 = torch.ones(4, requires_grad=True)
-        dynamo_out = torch._dynamo.optimize("aot_eager")(mod)(x2)
-        dynamo_out[0].backward(torch.ones(4))
+        dynamo_out = torch._dynamo.optimize("aot_eager", nopython=True)(mod)(x2)
+        with compiled_autograd.enable(compiler_fn):
+            dynamo_out[0].backward(torch.ones(4))
 
         self.assertEqual(dynamo_out, aot_out)
         self.assertEqual(dynamo_out, eager_out)
 
         self.assertEqual(x0.grad, x1.grad)
         self.assertEqual(x0.grad, x2.grad)
 
+    def test_input_hooks_same(self):
+        backends = ["eager", "aot_eager", "inductor"]
+        for backend in backends:
+
+            def my_hook(grad, *, k=0):
+                return grad + k
+
+            hook = functools.partial(my_hook, k=3)
```

Comment:
this pattern never actually happens in fsdp right

Reply:
Yea